packages/flutter/lib/src/rendering/shifted_box.dart

```diff
@@ -608,6 +610,20 @@ class RenderConstrainedOverflowBox extends RenderAligningShiftedBox {
     markNeedsLayout();
   }
 
+  /// This only affects scenario when the child does not indeed overflow.
+  /// If enabled, the self size will be as small as its child when the child
+  /// does not overflow. If disabled (the default case), the self size will
+  /// always be the biggest possible size that the parent constraints allows.
```

Comment:
"If set to true, the render object will size itself to match the size of its child within the constraints of its parent or be as small as the parent allows if no child is set. If set to false (the default), the render object will size itself to be as large as the parent allows."